program/src/state.rs

```diff
@@ -179,6 +216,48 @@ impl Config {
     }
 }
 
+/// An account metadata for a proposal instruction.
+#[derive(BorshDeserialize, BorshSerialize, Clone, Debug, Default, PartialEq)]
+pub struct ProposalAccountMeta {
+    /// The pubkey of the account.
+    pub pubkey: Pubkey,
+    /// Whether the account is a signer.
+    pub is_signer: bool,
+    /// Whether the account is writable.
+    pub is_writable: bool,
+}
+
+/// An instruction to be executed by a governance proposal.
+#[derive(BorshDeserialize, BorshSerialize, Clone, Debug, Default, PartialEq)]
+pub struct ProposalInstruction {
+    /// The program ID to invoke.
+    pub program_id: Pubkey,
+    /// The accounts to pass to the program.
+    pub accounts: Vec<ProposalAccountMeta>,
+    /// The data to pass to the program.
+    pub data: Vec<u8>,
+    /// Whether the instruction has been executed.
+    pub executed: bool,
```

Comment:
Not for this PR, but when executing, is the idea to check that all *previous* instructions have been executed?

Reply:
Yep! For example, if the parent wasn't executed, abort.